shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -181,7 +200,12 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
     return NO;
   }
 
-  const FlutterRendererConfig rendererConfig = [_openGLRenderer createRendererConfig];
+  FlutterRendererConfig rendererConfig;
+  if (_enableMetalRendering) {
+    rendererConfig = [_metalRenderer createRendererConfig];
+  } else {
+    rendererConfig = [_openGLRenderer createRendererConfig];
```

Comment:
The repeated use of this pattern in this class is an indication that these renderers should be using polymorphism. I would expect a single renderer ivar typed by a protocol that both versions implement.